File: freelance-system/src/main/java/com/example/freelance_system/model/WithdrawalRequest.java
package com.example.freelance_system.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "withdrawals")
public class WithdrawalRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long freelancerId;

    private String method;        // BANK, PAYPAL, PAYONEER etc.
    private String accountInfo;   // JSON String (email/account# etc.)

    private double amount;

    private String status = "COMPLETED";

    private LocalDateTime requestDate = LocalDateTime.now();

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getFreelancerId() {
		return freelancerId;
	}

	public void setFreelancerId(Long freelancerId) {
		this.freelancerId = freelancerId;
	}

	public String getMethod() {
		return method;
	}

	public void setMethod(String method) {
		this.method = method;
	}

	public String getAccountInfo() {
		return accountInfo;
	}

	public void setAccountInfo(String accountInfo) {
		this.accountInfo = accountInfo;
	}

	public double getAmount() {
		return amount;
	}

	public void setAmount(double amount) {
		this.amount = amount;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public LocalDateTime getRequestDate() {
		return requestDate;
	}

	public void setRequestDate(LocalDateTime requestDate) {
		this.requestDate = requestDate;
	}

   
}

File: freelance_app/lib/models/withdrawal_request.dart
class WithdrawalRequest {
  final int? id;
  final int freelancerId;
  final String method;
  final String accountInfo;
  final double amount;
  final String status;
  final DateTime requestDate;

  WithdrawalRequest({
    this.id,
    required this.freelancerId,
    required this.method,
    required this.accountInfo,
    required this.amount,
    this.status = "COMPLETED",
    required this.requestDate,
  });

  factory WithdrawalRequest.fromJson(Map<String, dynamic> json) {
    return WithdrawalRequest(
      id: json['id'],
      freelancerId: json['freelancerId'],
      method: json['method'],
      accountInfo: json['accountInfo'],
      amount: (json['amount'] as num).toDouble(),
      status: json['status'],
      requestDate: DateTime.parse(json['requestDate']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'freelancerId': freelancerId,
      'method': method,
      'accountInfo': accountInfo,
      'amount': amount,
      'status': status,
      'requestDate': requestDate.toIso8601String(),
    };
  }
}
